File: src/main/java/com/vocalclerk/api/dtos/UserSettingsDto.java
package com.vocalclerk.api.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for user settings data.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserSettingsDto {
    
    private Long id;
    private boolean darkMode;
    private boolean notificationsEnabled;
    private String preferredLanguage;
    private String aiModel;
    private boolean biometricAuthEnabled;
    private boolean dataBackupEnabled;
    private int reminderLeadTime;
}

File: backend/DoNow.Application/DTOs/UserSettingsDto.cs
namespace DoNow.Application.DTOs;

public class UserSettingsDto
{
    public long Id { get; set; }
    public bool DarkMode { get; set; }
    public bool NotificationsEnabled { get; set; }
    public string PreferredLanguage { get; set; } = "en";
    public string AiModel { get; set; } = "ollama";
    public bool BiometricAuthEnabled { get; set; }
    public bool DataBackupEnabled { get; set; }
    public int ReminderLeadTime { get; set; }
}
